arangod/Aql/Query.cpp

```diff
@@ -586,15 +587,13 @@ ExecutionState Query::execute(QueryRegistry* registry, QueryResult& queryResult)
             ExecContext const& exec = ExecContext::current();
 
             // got a result from the query cache
-            if (!exec.isSuperuser()) {
```

Comment:
If the user is a superuser then the user should have access to the collection. Therefore removing this short-cut as Superuser is not well-defined.